account_wallet/models/account_wallet.py

```diff
@@ -0,0 +1,116 @@
+# © 2015  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountWallet(models.Model):
+    _name = 'account.wallet'
+    _inherit = ['portal.mixin', 'mail.thread', 'mail.activity.mixin']
+    _description = "Wallet"
+    _check_company_auto = True
+
+    name = fields.Char(
+        readonly=True,
+        copy=False,
+        default=lambda self: _("New"),
+    )
+    wallet_type_id = fields.Many2one(
+        'account.wallet.type',
+        'Wallet Type',
+        required=True,
+        ondelete='restrict')
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Partner",
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        related="wallet_type_id.company_id",
+        store=True,
+    )
+    company_currency_id = fields.Many2one(
+        related='company_id.currency_id',
+        readonly=True,
+    )
+    balance = fields.Monetary(
+        compute='_compute_balance',
+        currency_field='company_currency_id',
+        store=True,
+        tracking=10
+    )
+    active = fields.Boolean(
+        default=True)
+    account_move_line_ids = fields.One2many(
+        comodel_name="account.move.line",
+        inverse_name="account_wallet_id",
+        string="Journal Items")
+    # TODO: Check if this is necessary
+    # create_date = fields.Date(
```

Comment:
create_date it's a magic field added by the ORM, no need to add it

Reply:
It was added in view (the field was there I don't know why - let it as TODO for record)